react/src/data/report.ts

```diff
@@ -1,49 +1,96 @@
-import { getSimulationFrame, pollRunReport, runStatus } from "../utility/compute";
+import { getSimulationFrame, pollRunReport, ResponseHasState, getRunStatusOnce } from "../utility/compute";
 import { v4 as uuidv4 } from 'uuid';
 import React from "react";
 import { ModelStates } from "../store/models";
+import { mapProperties } from "../utility/object";
 
 export const CReportEventManager = React.createContext<ReportEventManager>(undefined);
 
-export type ReportEventSubscriber = (simulationData: any) => void;
+export type ReportEventSubscriber = {
+    /**
+     * Called once for every simulationData received (including during completion)
+     */
+    onReportData?: (simulationData: ResponseHasState) => void;
+    /**
+     * Called once when reports are starting to be generated
+     */
+    onStart?: () => void;
+    /**
+     * Called once when reports are finished being generated
+     */
+    onComplete?: () => void;
+}
+
+type RunStatusParams = {
```

Comment:
These params are returned from the server in nextRequest from runSimulation and runStatus. They should be treated opaquely by the client. nextRequestSeconds is also returned, and that should be used for the poll on the next polling. Different simulations return different amounts.

Please look at the [angular code](https://github.com/radiasoft/sirepo/blob/8fb85dc7befa7684ac4de4631b053c75d47a014a/sirepo/package_data/static/js/sirepo.js#L2558) to see how to handle requests from the server.

Reply:
In order to "pick up where we left off" on reports during a page refresh these are sometimes manually provided. The client should also probably know about these so we know which requests can share parameters (like when running a report also begins polling).

Reply:
This isn't how the current app works. It simply calls runSimulation with all the data and the server decides if the simulation needs to run. This same logic could be used.

Reply:
Would this be a case where forceRun=false?

Reply:
Correct. The forceRun=True is for manual starts.

Reply:
It looks like the nextRequestSeconds has been addressed, so I'll resolve this item.